### ISSUE TITLE
App crash when extracting archive

### ISSUE THREAD

**Issue by eweap** [driftyco/ionic-plugins-deploy#12]
When deploying some updates, the deploy service download and start extracting the archive then the app crash with error:

```
Application Error

The file or directory could not be found. (file:///data/data/com.compagny.appname/app_339070a214f611e5ae823e00dea088b8/index.html)
```

Reinstall the app and clear cache through Settings/Applications, doesn't help.
The crash happen on Android 5.1.1, I didn't test on iOS for the moment.

There is my informations:
```
$ ionic info

Your system information:

Cordova CLI: 5.0.0
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.0
Ionic CLI Version: 1.5.4
Ionic App Lib Version: 0.2.1
OS: Distributor ID:	Ubuntu Description:	Ubuntu 14.04.2 LTS 
Node Version: v0.12.2
```

Any idea ? Thanks by advance !

**Comment by eweap** [comment]
I made a gist with my logcat: https://gist.github.com/eweap/e81bba67611542069d8a

**Comment by gaguirre** [comment]
+1

**Comment by wgjtyu** [comment]
at line 890

I/IONIC.DEPLOY.UNZIP_STEP(32672): Exception: /data/data/com.alhena.planetagger/app_339070a214f611e5ae823e00dba088b8/css/main.css: open failed: ENOTDIR (Not a directory)

it is because `css` folder was not created

**Comment by crazystars** [comment]
There is a problem with the unzip function. 
During the while in line 429  there is no check if the entry is a dir or not.
i' put in a simple if else 
just look at https://gist.github.com/crazystars/9fb26ca1966e6db24317
it is working for me now

**Comment by eweap** [comment]
@crazystars Thanks man ! It's working like a charm !

**Comment by affinionmobile** [comment]
Your pull request fixed my issue on Android. Deploy works now.

**Comment by ericb** [comment]
Hey guys, this explanation might seem a bit confusing but here goes:

This is actually an issue with Ionic CLI 1.5.1. When you run ionic upload, we generate and upload a zip file -- this is what the plugin eventually downloads. In 1.5.1 an upgrade to the node archiver was included. Sparing all the details, it produces a semi-borked version of the zip that Android's ZipInputStream fails on. 

We currently have a beta with the fix in place. you can test this new beta by installing `npm install ionic@beta`. The fix will make its way into a 1.6.x release.

Note: Because it's the zip file itself that is slightly corrupt, you'll need to upload a new version of your app to deploy and download on your device *after* you upgrade your CLI.

**Comment by eweap** [comment]
@ericb Great, the `ionic@beta` fix the problem ! Thanks man (and Ionic team) :)
